platforms/core-configuration/kotlin-dsl-provider-plugins/src/main/kotlin/org/gradle/kotlin/dsl/provider/plugins/precompiled/tasks/GeneratePrecompiledScriptPluginAccessors.kt

```diff
@@ -106,6 +121,12 @@ abstract class GeneratePrecompiledScriptPluginAccessors @Inject internal constru
     @get:Classpath
     abstract val runtimeClassPathFiles: ConfigurableFileCollection
```

Comment:
💭  It's a pity we don't have a way to go from `Provider<FileCollection>` to `FileCollection` in a lazy manner, so we can replace this with an actual getter to derive it from `runtimeClassPathArtifactCollection`.

🤔  Maybe using proper getter, setter, and a backing field instead of using a `Property` for `ArtifactCollection` would be cleaner in this regard? `ArtifactCollection` is a lazy type, after all. Then we can expose read-only derived `FileCollection`. I don't insist, of course.

Reply:
I think I would still keep a Property, just to avoid having a setter. 
But it's a good idea to have just a backing field and `abstract val runtimeClassPathFiles: ConfigurableFileCollection` is just a getter